### PR TITLE
pkg/util/{cgroups,log,scrubber}: migrate package to submodule

### PR DESCRIPTION
**Pull request by knusbaum**
### What does this PR do?

This change moves `pkg/util/cgoups`, `pkg/util/log`, and `pkg/util/scrubber` into their own submodules for use in other modules.

`pkg/util/cgroups` is needed for implementing #11584 without importing `github.com/DataDog/datadog-agent`.

`pkg/util/cgroups` depends on `pkg/util/log`, which in turn depends on `pkg/util/scrubber`. Out of necessity, these two packages have also been pulled into their own individual submodules.

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the `k8s/<min-version>` label, indicating the lowest Kubernetes version compatible with this feature. 
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

### CODE REVIEW

File: pkg/util/cgroups/pid_mapper.go [hunk 1]
```diff
@@ -107,6 +107,18 @@ func getPidMapper(procPath, cgroupRoot, baseController string, filter ReaderFilt
 	return pidMapper
 }
 
+// GetProcessNamespaceInode performs a stat() call on /proc/<pid>/ns/<namespace>
```
Comment (vboulineau) [inline pkg/util/cgroups/pid_mapper.go:110]: I have another PR that will require this function in `system`. Ideally we'd want to keep in a single place, but we'll need to duplicate for now.
I think we should add a comment here and leave the original copy in `system`.

File: pkg/util/cgroups/pid_mapper.go [hunk 2]
```diff
@@ -107,6 +107,19 @@ func getPidMapper(procPath, cgroupRoot, baseController string, filter ReaderFilt
 	return pidMapper
 }
 
+// getProcessNamespaceInode performs a stat() call on /proc/<pid>/ns/<namespace>
+// This has been copied from pkg/util/system/namespace_linux.go's GetProcessNamespaceInode
+func getProcessNamespaceInode(procPath string, pid string, namespace string) (uint64, error) {
```
Comment (vboulineau) [inline pkg/util/cgroups/pid_mapper.go:112]: That should move to `namespace_linux.go`, the comment should state that the whole code should in `pkg/util/system/namespace_linux.go` but currently isn't due to isolation requirement made necessary by OpenTelemetry depending on the `trace-agent`.

**Review by djmitche — APPROVED** [review]
pkg/{util,log} changes look good!

**Review by ahmed-mez — APPROVED** [review]
LGTM but let's wait for a +1 from @vboulineau as well!

**Review by vboulineau — APPROVED** [review]

**Review by hmahmood — APPROVED** [review]

**Pull request merged**